lib/bsdlib/bsdlib.c

```diff
@@ -25,9 +25,8 @@ static int _bsdlib_init(struct device *unused)
 	 */
 	IRQ_DIRECT_CONNECT(BSD_NETWORK_IRQ, BSD_NETWORK_IRQ_PRIORITY,
 			   ipc_proxy_irq_handler, 0);
-	bsd_init();
 
-	return 0;
+	return bsd_init();
```

Comment:
The return code from `bsd_init` lies in the `NRF_*` error space - I don't think we should propagate it further to Zephyr. Perhaps we should log the actual error code, and return something from Zephyr error space?

Reply:
The application uses the return value of `bsd_init` to determine if DFU was successful (or not).
At the moment it returns either zero, or one of the errors defined in `bsd.h`.

Reply:
I see your point now. At the very minimum, we should add more specific documentation for `bsdlib_init`, to indicate where to look for the error codes it can return.
 
And I agree it could make sense to split `bsdlib_init` and `_bsdlib_init` - the former one allowed to return custom error codes, the latter, used by `SYS_INIT`, not.

Reply:
So I have updated the code as discussed and will prepare a PR to `nrfxlib` with updated documentation for `bsd_init()` in `bsd.h`. Next, will open a PR to `fw-nrfconnect-nrf` to update `bsdlib_init()` documentation in `bsdlib.h`.